### DISCUSSION TITLE
Unique Product Link

### DISCUSSION THREAD

**Discussion by ghost** [btcpayserver/btcpayserver#2828] [Ideas / Feature Requests]
Right now I can only link to the entire Store, with all its products. 

It would be cool, if the visitor can generate a unique link for each of the products in that store. So that this link can be forwarded to other potential visitors with "checkout THIS SPECIFIC product".

**Comment by dstrukt** [comment]
I like the idea. 

Do you imagine an option for each item accessible via the list view, or a detail page for the item with the option as a possible option? I think most are more familiar with a UX closer to the latter, sans cart, etc... but curious to hear your thoughts.

"Share this store" VS "Share this item for checkout"

**Reply by ghost** [reply]
"Share this store" is just sending the link to the app, no?

I'm looking for "Share this item". 

But not sure if it should lead to a "details page" [nice suggestion], or put the item directly into the cart - proceed to checkout.

**Reply by bolatovumar** [reply]
The way it was described in the original message above it seemed to me like it meant we would have a details page for each item which you could link. This would certainly be doable but it does extend the scope of the POS app quite a bit. We would probably also need design work for how we would want the details page to look and work. 

Having the option of sharing the link to the existing POS app with some sort of query string parameter which just places a particular item into the cart when the page is loaded would be a lot simpler.

I'm curious as to what others think about this though.

**Reply by ghost** [reply]
good points.

But how would your second approach work, if there is no cart view, but just products? You cannot add them automagically in the cart...

**Reply by dstrukt** [reply]
I'll mock up a quick design to explain my thinking when I get a moment this weekend / will give us something more concrete to discuss.

**Comment by dstrukt** [comment]
Apologies for the delay, finally got a moment to mock up some quick wireframes.

Happy to iterate based on feedback, as these were just intended to get the ball rolling with some details ... assuming I understood everyone's concepts.

We have our standard "List View" with all items, I've opted to only show one, so the discussion stays focused, but couldn't help adding a few small elements that feed desperately needed for our core POS app. 
- Might be worth adding a "Share" button for the store level.
- Might be worth considering a cover image & description for the store.
- Might be worth considering a custom logo upload for the store.
<img width="808" alt="Screen Shot 2021-09-14 at 11 55 01 PM" src="https://user-images.githubusercontent.com/6250771/133385329-ad5b114e-1278-49e3-8b5e-4b22a6a8176b.png">

1. The most simple option being some form of a "Share" overlay, that might help provide some context + a copy/paste generated link that would potentially generate the checkout invoice upon load ... ? There are some UX and context concerns around this approach, but it's certainly a lower effort solution.
<img width="1444" alt="Screen Shot 2021-09-14 at 11 55 07 PM" src="https://user-images.githubusercontent.com/6250771/133385336-175f3b70-634d-435b-abff-76a6693eea7e.png">

2. The more complex cart option (with the obv. eng. effort required) ... allows multiple items added to the cart + ability to share this custom cart vs. just one item. Additionally this is a more common usability pattern for shoppers, but they might not understand they could share the cart, so we'd have to explain that. 

Will likely update this view as I'm in the process of improving the UI for this view, but left it wireframe for now.

<img width="1435" alt="Screen Shot 2021-09-14 at 11 55 14 PM" src="https://user-images.githubusercontent.com/6250771/133385339-d45ce641-503c-481d-bc9f-d2f8921c886b.png">

3. Detail view with the same options / linkable URL + share button (would just drop you on this unique product URL, and you'd generate an invoice normally).
<img width="1441" alt="Screen Shot 2021-09-14 at 11 55 20 PM" src="https://user-images.githubusercontent.com/6250771/133385342-4f8aa8f7-bfc5-4599-aad1-0c6444ee9851.png">

Am I missing anything? Just LMK and I'll do another quick pass tomorrow, but this should give us something concrete.

cc @MaxHillebrand @bolatovumar

**Reply by ghost** [reply]
Beautiful, thanks @dstrukt! All three are good solutions, 2 and 3 are pretty much perfect.
For 1, do we really need this popup dialog with the url visible? It might be sufficient to simply have the `Share` button in the product, which auto-copies the link. I'm not sure what the link should include for this though. So I'd say, that 3 is the more holistic approach, so maybe, 1 `Share` button can include the link to details page 3.

**Reply by bolatovumar** [reply]
> 1. The most simple option being some form of a "Share" overlay, that might help provide some context + a copy/paste generated link that would potentially generate the checkout invoice upon load ... ? There are some UX and context concerns around this approach, but it's certainly a lower effort solution.

Yeah, this would probably be the simplest but it may not make sense  to the receiver that they just get an invoice and don't get to see any details about the product?

> 2. The more complex cart option (with the obv. eng. effort required) ... allows multiple items added to the cart + ability to share this custom cart vs. just one item. Additionally this is a more common usability pattern for shoppers, but they might not understand they could share the cart, so we'd have to explain that.

I like this, it makes sense and it's not mutually exclusive with option 3. We could have both really.

> 3. Detail view with the same options / linkable URL + share button (would just drop you on this unique product URL, and you'd generate an invoice normally).

I like this as well and I'm wondering if we would allow customizing this page or not. I'm thinking we could start simple and just have the content fixed based on what's already available but it could be made customizable later.
